pkg/upgrade/job/job.go

```diff
@@ -264,7 +264,11 @@ func New(plan *upgradeapiv1.Plan, node *corev1.Node, controllerName string) *bat
 			//only available in kubectl version 1.18 or later
 			args = append(args, "--skip-wait-for-delete-timeout", strconv.FormatInt(int64(drain.SkipWaitForDeleteTimeout), 10))
 		}
-
+		if drain.PodSelector != nil {
+			args = append(args, "--pod-selector", `!`+upgradeapi.LabelController+`,`+*drain.PodSelector)
+		} else {
+			args = append(args, "--pod-selector", `!`+upgradeapi.LabelController)
+		}
```

Comment:
I haven't touched Go code in nearly a year but this looks like it could be a one-liner to me. 

Thank you for remembering the magic that prevents the upgrade from draining itself off of the node-to-be-upgraded!